File: 2015/02-20/Pat/Challenge1.rb
#!/usr/bin/env ruby
#
# Author:      Pat Hawks
# Created on:  Feb 02, 2015
# Source File: Challenge1.rb
#
# To run:
#     ruby Challenge1.rb

MIN_NUM = 1000   # Ruby does not actually enforce constancy, but will produce a
MAX_NUM = 999999 # warning if these variables are later modified
PROMPT_STRING = """\
Enter a whole number between #{MIN_NUM} and #{MAX_NUM} without spaces or commas.
"""
FORMAT_STRING = "%d,%.3d\n"

begin
    puts PROMPT_STRING
    @inNum = gets.to_i
end while @inNum < MIN_NUM or @inNum > MAX_NUM

puts FORMAT_STRING%[ @inNum/1000, @inNum%1000 ]

File: 2015/02-20/Pat/Challenge1.py
#!/usr/bin/env python
#
# Author:      Pat Hawks
# Created on:  Feb 02, 2015
# Source File: Challenge1.py
#
# To run:
#     python Challenge1.py

MIN_NUM = 1000   # Python does not actually enforce constancy, but hopefully the
MAX_NUM = 999999 # SCREAMING_SNAKE_CASE is a hint to develoopers not to modify
                 # these variables
PROMPT_STRING = '''\
Enter a whole number between {min} and {max} without spaces or commas.
'''.format( min=MIN_NUM, max=MAX_NUM )
FORMAT_STRING = "{:,}"
inNum = None

while True: # What I'm really after is a `do while` loop
    try:
        inNum = int( input( PROMPT_STRING ) )
        if inNum >= MIN_NUM and inNum <= MAX_NUM:
            break
    except: # If the input cannot be converted to an int:
        pass

print( FORMAT_STRING.format( inNum ) )
